agent/custom/general_navigation_action.py

```
@@ -1,238 +1,482 @@
+            return any(tx >= nx + nw * 0.4 and tx <= nx + nw + 5 and ty < ny + nh and ty + th > ny
+                       for tx, ty, tw, th, tc in tags)
+
+        def paired_t(tx, ty, tw, th):
+            return any(tx >= nx + nw * 0.4 and tx <= nx + nw + 5 and ty < ny + nh and ty + th > ny
+                       for nx, ny, nw, nh, nc in nametags)
+
+        un_n = [(x, y, w, h) for x, y, w, h, c in nametags if not paired_n(x, y, w, h)]
+        un_t = [(x, y, w, h) for x, y, w, h, c in tags if not paired_t(x, y, w, h)]
+        # 孤 nametag 向右窗口重检找 tag
+        for box in un_n:
+            for (x1, y1, x2, y2, c, cl) in self._recheck(img, box, "right"):
```

Ruff (0.16.1)

[warning] Loop control variable `cl` not used within loop body

(B007)